source/AI.cpp

```diff
@@ -1978,7 +1980,21 @@ void AI::PrepareForHyperspace(Ship &ship, Command &command)
 	bool isJump = (ship.GetCheapestJumpType(ship.GetTargetSystem()).first == Ship::JumpType::JumpDrive);
 
 	Point direction = ship.GetTargetSystem()->Position() - ship.GetSystem()->Position();
-	if(!isJump && scramThreshold)
+	double squaredDeparture =  isJump ?
+		ship.GetSystem()->JumpDepartureDistance() * ship.GetSystem()->JumpDepartureDistance() :
+		ship.GetSystem()->HyperDepartureDistance() * ship.GetSystem()->HyperDepartureDistance();
+	squaredDeparture += SAFETY_OFFSET;
+	if(ship.Position().LengthSquared() < squaredDeparture)
+	{
+		Point closestDeparturePoint;
+		if(ship.Position())
+			closestDeparturePoint = ship.Position()
+				* (squaredDeparture / ship.Position().LengthSquared());
+		else
+			closestDeparturePoint = Point(1., 1.) * (squaredDeparture / Point(1., 1.).LengthSquared());
```

Comment:
No need to calculate a point here, it is possible to immediately generate one. (But it doesn't really matter much, this code gets rarely executed.)
```suggestion
			closestDeparturePoint = Point(1., squaredDeparture);
```